ryangray-ryanmgray/hw_3/car_lot.rb

```diff
@@ -0,0 +1,531 @@
+      puts "I am now $#{the_car[:price]} richer!"
+      return "sale"
+    else
+      puts "oh... ok then... (I was so close!!!! T_T)"
+      return "no sale"
+    end
+  else
+    puts "Sorry, that's not a valid ID."
+    get_id(cars)
+  end
+end
+
+
+
+#SALES DEPARTMENT
+#This will ask the user if they're willing to accept flexibility in their criteria, and will call the appropriate methods if the user agrees.
+def sales_department(year, price, elec, color, car_lot)
+  puts "Wait! We might have more matches for you if you have a little flexibiilty in your criteria!"
+  try_more = yes_no("Do you want to keep looking for cars?")
+
+  if try_more == true
+    new_match = flex_match(year, price, elec, color, car_lot)
+    if new_match.length > 0
+      ask_for_sale(new_match)
+    else
+      puts "We can't find any new matches for you."
+      sales_department(year, price, elec, color, car_lot)
+    end
+  else
+    puts "Alright, thanks anyways.... I guess.... "
+  end
+
+end
+
+#----------------------------------Sales----------------------------------
+
+
+
+
+#------------------------------Body of the code------------------------------
+
+#setting up the inventory.
+car_lot = []
+tesla = {ID: 1001,name: "Tesla", model: "Model X", year: 2017, is_electric: true, price: 45000, color: "black"}
+jeep = {ID: 1002,name: "Jeep", model: "Grand Cherokee", year: 2004, is_electric: false, price: 37000, color: "white"}
+porsche = {ID: 1003,name: "Porsche", model: "Speedster", year: 1955, is_electric: false, price: 65000, color: "yellow"}
+ford = {ID: 1004,name: "Ford", model: "Fusion", year: 2003, is_electric: false, price: 18000, color: "red"}
+toyota = {ID: 1005, name: "Toyota", model: "Prius", year: 2010, is_electric: true, price: 24000, color: "silver"}
+batmobile = {ID: 1006, name: "Batmobile", model: "BAT9000", year: 2017, is_electric: true, price: 250000, color: "black"}
+flintmobile = {ID: 1007, name: "Flintmobile", model: "RockWagon", year: 400, is_electric: false, price: 30, color: "red"}
+nimbus = {ID: 1008, name: "Nimbus", model: "Nimbus 2000", year: 1991, is_electric: false, price: 500, color: "orange"}
+honda = {ID: 1009, name: "Honda", model: "Civic", year: 2006, is_electric: true, price: 27000, color: "silver"}
+generallee = {ID: 1010, name: "General Lee", model: "Charger", year: 1969, is_electric: false, price: 18000, color: "orange"}
+bmw = {ID: 1011, name: "BMW", model: "M3", year: 1993, is_electric: false, price: 35000, color: "white"}
+
+#Code technically starts here.
+car_lot.push(tesla, jeep, porsche, ford, toyota, batmobile, flintmobile, nimbus, honda, generallee, bmw)
+
+user_criteria = {}
+
+puts "Please enter your criteria"
+user_criteria[:lowest_year] = get_year
+user_criteria[:price] = get_price
+user_criteria[:electric_pref] = is_electric
+user_criteria[:colors] = color_checker(car_lot)
+
+match(user_criteria, car_lot)
+
+#-----------------------------Body of the code end----------------------------
```

Comment:
This is seriously some of the best commenting I've ever seen, Ryan. Keep it up! 👍